scripts/magpietts/infer_and_evaluate.py

```diff
@@ -54,33 +73,36 @@ def run_inference(
         confidence_level=0.95,
         use_local_transformer=False
     ):
-    # import ipdb; ipdb.set_trace()
-    model_cfg = OmegaConf.load(hparams_file).cfg
+    # Load model
+    if hparams_file is not None:
+        model_cfg = OmegaConf.load(hparams_file).cfg
 
-    with open_dict(model_cfg):
-        model_cfg.codecmodel_path = codecmodel_path
-        if hasattr(model_cfg, 'text_tokenizer'):
-            # Backward compatibility for models trained with absolute paths in text_tokenizer
-            model_cfg.text_tokenizer.g2p.phoneme_dict = "scripts/tts_dataset_files/ipa_cmudict-0.7b_nv23.01.txt"
-            model_cfg.text_tokenizer.g2p.heteronyms = "scripts/tts_dataset_files/heteronyms-052722"
-            model_cfg.text_tokenizer.g2p.phoneme_probability = 1.0
-        model_cfg.train_ds = None
-        model_cfg.validation_ds = None
+        with open_dict(model_cfg):
+            model_cfg = update_config(model_cfg, codecmodel_path)
 
+        model = MagpieTTS_Model(cfg=model_cfg)
+        model.use_kv_cache_for_inference = True
 
-    model = MagpieTTSModel(cfg=model_cfg)
-    model.use_kv_cache_for_inference = True
+        # Load weights from checkpoint file
+        print("Loading weights from checkpoint")
+        ckpt = torch.load(checkpoint_file, weights_only=False)
+        model.load_state_dict(ckpt['state_dict'])
+        checkpoint_name = checkpoint_file.split("/")[-1].split(".ckpt")[0]
+    elif nemo_file is not None:
+        model_cfg = MagpieTTS_Model.restore_from(nemo_file, return_config=True)
+        with open_dict(model_cfg):
+            model_cfg = update_config(model_cfg, codecmodel_path)
+        model = MagpieTTS_Model.restore_from(nemo_file, override_config_path=model_cfg)
+        model.use_kv_cache_for_inference = True
+        checkpoint_name = nemo_file.split(".nemo")[0]
+    else:
+        raise ValueError("Need a checkpoint")
 
-    # Load weights from checkpoint file
-    print("Loading weights from checkpoint")
-    ckpt = torch.load(checkpoint_file, weights_only=False)
-    model.load_state_dict(ckpt['state_dict'])
     print("Loaded weights.")
     model.cuda()
```

Comment:
is it necessary? if calling `ModelPT.restore_from()` without specifying `map_location=cpu`, the model should be cuda ready.